lib/class.OS_Linux.php

```diff
@@ -935,18 +935,19 @@ public function getBattery() {
 		// Get vals for each battery
 		foreach ($bats as $b) {
 
-			$go_for_it = true;
-
-			foreach(array($b.'/manufacturer', $b.'/status', $b.'/charge_now') as $f)
-				if (!is_file($f))
-					$go_for_it = false; // Continue out of two nested loops
-
-			if (!$go_for_it) 
-				continue;
+			foreach(array($b.'/manufacturer', $b.'/status') as $f)
+				if (!is_file($f))continue;
```

Comment:
The point of this is to break out of the $bats loop if either of those files are missing. The original code satisfies this; to do it with smaller code, you'll need to do "continue 2;" instead.

Also don't put the continue on the same line as the if; normal indention applies.